src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -160,9 +150,9 @@ object DepositHandler {
     }
   }
 
-  def checkBagStoreBaseDir()(implicit id: String, bs: BagStoreBase): Try[Unit] = {
-    if (bs.isBagStoreAware) {
-      val baseDir = new File(bs.baseDir)
+  def checkBagStoreBaseDir()(implicit id: String, bs: Option[BagStoreSettings]): Try[Unit] = {
+    if (bs.nonEmpty) {
```

Comment:
I think `checkBagStoreBaseDir` only is supposed to be called whenever the bagstore is actually configured. So you can leave out the `Option` part completely and move that to `finalizeDeposit` (see https://github.com/vesaakerman/easy-sword2/pull/4/files#diff-ad17b49d50f1b93808fb326e8f79d0d8R87).